FLEDGE_implementation_overview.md

```diff
@@ -0,0 +1,66 @@
+
+### Scoring Bids
+Once the seller executor has been loaded, of which there is always only one per AuctonConfig, and Promises resolved, bids are passed to scoreAd() as soon as they’re available, on a first-come-first-served basis.  If any additionalBids are passed in to an auction, they’ll likely be scored first.  Since ties are resolved randomly, and there’s currently no timeouts that span multiple scoreAd() calls, the order in which bids are scored currently does not matter.  Each scoreAd() call independently respects the scoreAd() timeout, which includes the time to run the Javascript file as well as the time to run scoreAd().
+
+Trusted seller signals fetches are started every 100 milliseconds, when there’s a bid a seller executor is waiting to score.  Once the final bid for an auction has been generated, the final signals fetch is immediately issued.  As with bidders, seller executors can be shared between auctions in the same frame running concurrently, which also allows for for shared seller signals fetches.
+
+Once all bids are scored, the result is immediately returned to the Javascript Promise.
+
+## Phase 3 - Reporting Scripts
+On auction completion, Chrome immediately reuses the seller executor to call reportResult(), and then reloads the winning bidder’s executor to call reportWin(), if it’s not still in memory.  The winning bidder executor is not kept in memory to avoid consuming bidder process quota, and to potentially avoid causing deadlock (e..g, if there are 10 auctions, all keeping different currently top-scoring bidders around, but that still need to load more bidders before they know the final winner, no auction could proceed).
+
+Nothing is reported to the server, and the win is not recorded, until the winning ad is navigated to, or deprecatedURNToURL() is invoked, which may happen while still running reporting scripts.  If the ad loads before a reporting script completes, and tries to send a report that depends on reportWin() / reportResult(), and that function hasn’t completed, the report is queued until the script has finished running.  Reporting scripts have a default timeout of 50 milliseconds.  In general, the run duration of these should not be web visible, though if the frame the auction ran in is destroyed before they complete, any currently in-progress work will be aborted.  This is considered a bug that needs to be fixed.  Running reporting scripts can affect the performance of running auctions, since worklet processes used by reporting count towards the global Protected Audience process quotas.
```

Comment:
Do reporting functions also take up worklet process JavaScript thread time? IIUC, this would block scoreAd() / generateBid() from running for other auctions for that seller / buyer?

Reply:
Added comment about this.